src/main/mbar.c

```diff
@@ -324,7 +442,38 @@ static void metColorInit(void) {
     }
 }
 
-INCLUDE_ASM("asm/nonmatchings/main/mbar", metColorSet);
+static void metColorSet(EXAM_TYPE exam_type, float per) {
+	u_char *moto_pp;
+	u_char *saki_pp;
+	int sakiper;
+
+    if (per == 0.f) {
+        return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].df_num));
+    }
+    if (per == -1.f) {
+        return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].ng_num));
+    }
+    if (per == 1.f) {
+        return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].ok_num));
+    }
```

Comment:
This is just a style nitpick, but I prefer defining the decimals even when it's zero.
```c
if (per == 0.0f) {
    return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].df_num));
}
if (per == -1.0f) {
    return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].ng_num));
}
if (per == 1.0f) {
    return Tim2Trans(cmnfGetFileAdrs(metcol_str[exam_type].ok_num));
}
```